contracts/apps/TicTacToe.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity 0.4.24;
+pragma experimental "ABIEncoderV2";
+
+import "../lib/Transfer.sol";
+
+
+contract TicTacToe {
+
+  enum ActionType {
+    PLAY,
+    PLAY_AND_WIN,
+    PLAY_AND_DRAW,
+    DRAW
+  }
+
+  enum WinClaimType {
+    COL,
+    ROW,
+    DIAG,
+    CROSS_DIAG
+  }
+
+  struct WinClaim {
+    WinClaimType winClaimType;
+    uint256 winClaimIdx;
+  }
+
+  struct AppState {
+    address[2] players;
+    uint256 turnNum;
+    uint256 winner; // 0 => game in progress, 3 => drawn, i => player[i-1] won otherwise
+    uint256[3][3] board; // 0 => empty square, i => players[i-1] otherwise
+  }
+
+  struct Action {
+    ActionType actionType;
+    uint256 playX;
+    uint256 playY;
+    WinClaim winClaim;
+  }
+
+  function turn(AppState state)
+    public
+    pure
+    returns (uint256)
+  {
+    return state.turnNum % 2;
+  }
+
+  function applyAction(AppState state, Action action)
+    public
+    pure
+    returns (bytes)
+  {
+    AppState memory postState;
+    if (action.actionType == ActionType.PLAY) {
+      postState = playMove(state, state.turnNum % 2, action.playX, action.playY);
+    } else if (action.actionType == ActionType.PLAY_AND_WIN) {
+      postState = playMove(state, state.turnNum % 2, action.playX, action.playY);
+      assertWin(state.turnNum % 2, postState, action.winClaim);
+      postState.winner = (postState.turnNum % 2) + 1;
+    } else if (action.actionType == ActionType.PLAY_AND_DRAW) {
+      postState = playMove(state, state.turnNum % 2, action.playX, action.playY);
+      assertBoardIsFull(postState);
+      postState.winner = 3;
+    } else if (action.actionType == ActionType.DRAW) {
+      assertBoardIsFull(state);
+      postState = state;
+      postState.winner = 3;
+    }
+
+    postState.turnNum += 1;
+
+    return abi.encode(postState);
+  }
+
+  function resolve(AppState state, Transfer.Terms terms)
+    public
+    pure
+    returns (Transfer.Details)
+  {
+    require(state.winner != 0);
+
+    uint256[] memory amounts = new uint256[](2);
+    address[] memory to = new address[](2);
+    bytes memory data; // = 0
+
+    if (state.winner == 3) {
+      amounts[0] = terms.limit / 2;
+      amounts[1] = terms.limit / 2;
+
+      to[0] = state.players[0];
+      to[1] = state.players[1];
+
+      return Transfer.Details(
+        terms.assetType,
+        terms.token,
+        to,
+        amounts,
+        data
+      );
+
+    } else {
+      address loser = state.players[state.winner - 1];
+      address winner = state.players[2 - state.winner];
+
+      amounts[0] = terms.limit;
+      amounts[1] = 0;
+
+      to[0] = loser;
+      to[1] = winner;
```

Comment:
Wait, isn't this giving the full amount of the transfer to the `loser` and not the `winner`? 🤔

Reply:
good catch 😂